### PR TITLE
Make builds reproducible with the help of Krankerl

### PR DESCRIPTION
**Pull request by ChristophWurst**
To avoid https://github.com/nextcloud/contacts/issues/2488 and https://github.com/nextcloud/contacts/issues/2489 in the future.

Now you create a release package with `krankerl package`. Nothing else. No command order to stick to. No local changes that break your tarball.

This should make it harder for me to screw up releases. So the theory.

Test build 1: [contacts.tar.gz](https://github.com/nextcloud/contacts/files/7469665/contacts.tar.gz)
Test build 2: [contacts.tar.gz](https://github.com/nextcloud/contacts/files/7469761/contacts.tar.gz)

### CODE REVIEW

**Review by miaulalala — APPROVED** [review]

File: .github/workflows/package.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    name: Build release tarball
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Set up Node
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16
+      - name: Set up php
+        uses: shivammathur/setup-php@master
```
Comment (skjnldsv) [inline .github/workflows/package.yml:16]:
```suggestion
        uses: shivammathur/setup-php@v2
```

**Review by kesselb — APPROVED** [review]
Nice :+1:

File: .github/workflows/package.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      - name: Install Krankerl
+        run: |
+          wget https://github.com/ChristophWurst/krankerl/releases/download/v0.12.3/krankerl_0.12.3_amd64.deb
+          sudo dpkg -i krankerl_0.12.3_amd64.deb
+      - name: Package app
+        run: krankerl package
+      - uses: actions/upload-artifact@v2
+        with:
+          path: build/artifacts/contacts.tar.gz
+          if-no-files-found: error
```
Comment (skjnldsv) [inline .github/workflows/package.yml:30]:
```suggestion
name: Packaging
on: pull_request

jobs:
  release-tarball:
    runs-on: ubuntu-latest
    
    name: Build release tarball
    steps:
      - name: Checkout
        uses: actions/checkout@v2

      - name: Set up Node
        uses: actions/setup-node@v2
        with:
          node-version: 16

      - name: Set up php
        uses: shivammathur/setup-php@v2
        with:
          php-version: 8.0
          extensions: ctype,curl,dom,gd,iconv,intl,json,mbstring,openssl,posix,sqlite,xml,zip,gmp
          coverage: none

      - name: Install Krankerl
        run: |
          wget https://github.com/ChristophWurst/krankerl/releases/download/v0.12.3/krankerl_0.12.3_amd64.deb
          sudo dpkg -i krankerl_0.12.3_amd64.deb

      - name: Package app
        run: krankerl package

      - uses: actions/upload-artifact@v2
        with:
          path: build/artifacts/contacts.tar.gz
          if-no-files-found: error
```

File: .github/workflows/package.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+name: Packaging
+on: pull_request
```
Comment (skjnldsv) [inline .github/workflows/package.yml:2]: Not fond of abusing the CI to build on every PR :confused: 
Maybe only on stuff that touches the info.xml

```suggestion
on: 
  pull_request:
    paths: 
      - 'appinfo/info.xml'

```

File: krankerl.toml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+[package]
+before_cmds = [
+    "composer install --no-dev -o",
```
Comment (skjnldsv) [inline krankerl.toml:3]: I don't think it's needed as it's only for testing we have a composer file

**Review by skjnldsv — CHANGES_REQUESTED** [review]
See comments

File: .github/workflows/package.yml (hunk 3 above)
Comment (ChristophWurst) [reply .github/workflows/package.yml:2]: while it's very seldom there have been issues with the build process only caught with such a check in the Mail app repo

the purpose is not to create an artifact. it's more of a smoke test. it's run just to make sure nothing throws an error

**Comment by ChristophWurst** [comment]
conflicts and should be redone

**Pull request closed without merging**